conda_lock/src_parser/__init__.py

```diff
@@ -136,3 +138,85 @@ def aggregate_lock_specs(
         platforms=ordered_union(lock_spec.platforms or [] for lock_spec in lock_specs),
         sources=ordered_union(lock_spec.sources or [] for lock_spec in lock_specs),
     )
+
+
+def parse_source_files(
+    src_files: List[pathlib.Path],
+    platform_overrides: Optional[Sequence[str]],
+    pip_support: bool = True,
+) -> List[LockSpecification]:
+    """
+    Parse a sequence of dependency specifications from source files
+
+    Parameters
+    ----------
+    src_files :
+        Files to parse for dependencies
+    platform_overrides :
+        Target platforms to render environment.yaml and meta.yaml files for
+    """
+    from conda_lock.src_parser.environment_yaml import parse_environment_file
+    from conda_lock.src_parser.meta_yaml import parse_meta_yaml_file
+    from conda_lock.src_parser.pyproject_toml import parse_pyproject_toml
```

Comment:
Those modules import `conda_lock/src_parser/__init__.py`. When I included them at the top, I was getting a circular dependency error. Those files use the `SourceDependency`, `SourceFile`, `VersionedDependency`, and `URLDependency` classes. If I move these classes to a file like `conda_lock/src_parser/models.py`, then I can get rid of the circular dependency and have these be top level imports. Thoughts?

Reply:
It's not just for type hints. I will move them to a new module.

Reply:
I forgot to mention in my previous comment that Python does permit certain types of circular imports.

Running `from a import B` is essentially equivalent to `import c; B = c.B`. Python will allow you to import `c` from `a` while also importing `a` from `c`, as long as you don't access the module attributes (i.e. `c.B`) before the module has been fully loaded. (This works when all `a.B` accesses occur within function definitions.)

So there is another strategy: rather than fix the cycles, try to import modules lazily, i.e. replace `from c import B` → `import c` and `B` → `c.B`. But I have the impression that this leads to very fragile code. I think it's generally much more robust to remove all non-typing-related circular imports when possible.